### ISSUE TITLE
Generate frontier-assisted Mathia intuitions with tools allowed and output-only proof boundary

### ISSUE THREAD

**Issue by murillo128** [murillo128/mathia#59]
Research umbrella: #2
Conceptual-specialist line: #29
Historical frontier-reference protocol: #32
Prior failed attempt / immutable blocker evidence: #57 / PR #58
Proof-aware oracle condition (different experiment): #52
Mathia-v2 general design: #55
Downstream validation population: `murillo128/qwen-lean#78`
Future planner work: `murillo128/qwen-lean#80` — **out of scope here**

# Goal

Generate and freeze a complete **frontier-assisted Mathia-style intuition corpus** for the qwen-lean Dataset-v2 validation population.

The objective is **not to evaluate GPT-5.6 Sol as a clean model-only baseline**. The objective is to obtain the strongest practical mathematical intuitions we can produce from a frontier system and later ask whether a frozen qwen-lean worker can exploit them.

Conceptually:

```text
public validation theorem
        |
        v
GPT-5.6 Sol / xhigh
+ reasoning
+ tools / web / retrieval when useful
        |
        v
best available mathematical understanding
        |
        v
STRICT OUTPUT BOUNDARY
Mathia-style natural-language intuition only
```

A later, separate qwen-lean issue may compare:

```text
Q4 direct
frontier-assisted intuition -> Q4
Mathia-v2 intuition         -> Q4
```

This issue generates/finalizes the intuition corpus only. It does **not** run qwen-lean, DeepSeek, Lean verification, Mathia-v2 inference, planner inference/training, or RL.

# Why this supersedes #57 operationally

#57 is preserved as valid blocker evidence. Do not rewrite or relabel it.

Its independent audit found two distinct problems:

1. the runtime emitted tool/web items despite the frozen no-tools contract;
2. the deterministic eligibility gate still admitted some Lean-plan / formal-instruction / near-complete-proof outputs.

For the present scientific objective, finding (1) is **no longer considered a defect**. Tools, browsing, retrieval and other reasoning support are allowed because we are not trying to estimate isolated GPT capability.

Finding (2) **remains a material defect**. The downstream artifact must still represent mathematical intuition rather than transmit a Lean proof implementation.

Therefore this is a new experiment/version, not a repair or reinterpretation of the frozen #57 run.

# Exact workload — all 650 validation tasks

Use the complete frozen validation population, with no outcome-based filtering:

- `minif2f-valid-clean-v2`: **244 tasks**;
- `fresh-composition-valid-v2`: **406 tasks**;
- total: **650 tasks**.

Do not inspect or include final-test tasks.

Do not select tasks based on Q4 pass@k, `c_i`, solved/unsolved status, DeepSeek outcome, mathematical domain, or apparent difficulty. Later analysis may stratify the frozen full corpus, but generation membership is outcome-blind.

## Frozen qwen-lean lineage

Authoritative Dataset-v2 acceptance commit:

```text
murillo128/qwen-lean@67d0cdc13ce38c3847633fe9982d0c6e1473e8ed
```

Current accepted validation membership/order anchors from `murillo128/qwen-lean#78`:

```text
miniF2F ordered_task_ids_sha256:
cb5e5edba99dcb1fad61a1f5f88fd38e0061d122b54ea4e39a630a18138cce13

fresh-composition ordered_task_ids_sha256:
7d01d2878a996a0fc57df1eb634ba562cebe396002dc0bd1f617b3b2e7e80f0b
```

Materialize exactly those 650 validation tasks. If counts/hashes do not match, stop rather than generating against a different population.

# Information boundary — theorem-only with respect to privileged formal evidence

The frontier system may receive:

- the exact public theorem/declaration;
- public imports/preamble/context needed to understand its mathematical meaning;
- stable workload/task identity as non-semantic metadata;
- the frozen intuition-generation instruction.

It must **not** receive, directly or indirectly:

- source/oracle/reference proof;
- qwen-lean generated candidates;
- Lean verifier errors or outcomes;
- Q0/Q1/Q2/Q3/Q4 results;
- pass@k, `c_i`, solved/unsolved bucket;
- DeepSeek outcomes;
- #78 Stage-9 capability-gap labels;
- any hint derived from a known formal solution.

The distinction is important:

- **tools/web/retrieval are allowed**;
- **privileged benchmark proof/outcome data are not**.

If external sources happen to discuss a famous theorem or a mathematical method, that is acceptable for this frontier-assisted upper-bound condition. Do not deliberately search for or ingest the benchmark's exact Lean proof implementation.

This is still distinct from #52, where the verified source proof is intentionally visible to an oracle generator.

# Generator contract

Primary generator:

```text
model: GPT-5.6 Sol
reasoning effort: xhigh
fresh isolated session per theorem
conversation state not reused across tasks
```

Tools are explicitly permitted, including where available:

- web search/browsing;
- retrieval/reference lookup;
- calculator/computation;
- note/todo/reasoning helpers;
- other non-privileged tools useful for understanding the mathematics.

Record every tool call/use in raw provenance, but **tool use must not itself make an output ineligible**.

Do not make tool availability identical across tasks by artificial padding. The generator may use tools adaptively when useful.

# Intuition-generation instruction

The instruction should strongly separate **understanding** from **formal execution**. Freeze exact wording before the calibration run, but it should convey the following semantics:

> Analyze the theorem and identify the key mathematical idea that makes it true. Give a compact intuition explaining the useful representation, invariant, decomposition, obstruction, symmetry, reduction, or conceptual mechanism. You may state a small number of intermediate mathematical observations if they clarify the route. Focus on *why the argument should work*, not on how to encode it in Lean. Do not provide Lean tactics, mathlib lemma names, formal proof code, or a step-by-step complete proof.

The desired artifact is something a mathematical collaborator could tell a formal prover before proof search.

Examples of acceptable content:

- “Rewrite the expression as a difference of squares; the hypothesis then fixes the symmetric factor and reduces the remaining claim to a polynomial identity.”
- “Use parity rather than direct divisibility: the recurrence preserves oddness modulo 2, after which the target divisibility follows from the factorization.”
- “View the statement as monotonicity of a convex quantity; Jensen/AM-GM supplies the missing inequality once the normalization is exposed.”

The examples above are illustrative only; do not include benchmark-specific solution content in the frozen generator prompt unless independently justified.

# Output boundary — the only strict interface gate

The final downstream-visible intuition must be **natural-language mathematical guidance**, not a formal proof implementation.

## Allowed

- ordinary mathematical notation and equations;
- names of standard mathematical concepts/theorems when used conceptually;
- representations, invariants, symmetry, contradiction, induction, decomposition, case structure;
- a small number of conceptual intermediate goals;
- qualitative advice such as “reduce to a polynomial identity” or “use induction on n”;
- explicit statement of an essential assumption/obstruction.

## Disallowed

- Lean code, code fences containing formal code, `:= by`, tactic scripts or proof terms;
- exact mathlib-qualified identifiers or API recipes intended as implementation instructions;
- sequences such as “apply X; rw [Y]; exact Z” or equivalent prose transcription;
- copied formal hypothesis names/identifiers used as a Lean execution recipe;
- a near-complete step-by-step proof whose remaining formalization is mechanical transcription;
- the source/oracle proof or fragments derived from it.

Generic tactic-family words such as “induction”, “normalization”, “ring arithmetic”, or “linear arithmetic” are **not automatically disallowed** when they are part of natural mathematical guidance. The gate should reject transmission of an implementation, not ordinary mathematical vocabulary.

Do not reject merely because the word `Lean` appears; instead make generation prompt/output policy prevent Lean-oriented instructions and classify the actual semantic boundary conservatively.

# Length contract — recalibrate rather than inheriting 96 blindly

The 96-token cap in #57 came from an older intuition-fertility interface and caused substantial rejection pressure. This new corpus is intended to capture the best compact intuition, not to reproduce that historical token budget blindly.

Before the full run, perform a **small calibration set of 24 tasks** drawn deterministically before seeing outputs:

- 8 miniF2F;
- 16 fresh composition;
- spread across the frozen task ordering, not selected by known model difficulty.

Use calibration only to verify interface quality and choose/freeze a compact maximum output budget from these predeclared options:

```text
128 / 160 / 192 qwen-lean-tokenizer tokens
```

Selection rule:

- choose the smallest cap under which at least 22/24 calibration outputs satisfy the semantic output boundary **without semantic truncation**;
- if none qualifies, stop with a design blocker instead of increasing the cap post hoc.

Do not score the mathematical usefulness of the 24 against qwen-lean/Lean. Calibration is only for output-boundary/length viability.

Once chosen, freeze the cap before generating the remaining tasks. Do not truncate outputs to fit; regenerate under the same prompt with an explicit compactness reminder according to the retry policy below.

# Eligibility / validation architecture

The #57 rule gate was both too broad in some places and incomplete in others. This issue should use a **two-layer output-boundary check**:

1. deterministic hard checks for unambiguous formal leakage (code blocks/formal syntax/tactic-script patterns/qualified Lean identifiers/etc.);
2. a separate blinded semantic boundary review for cases not mechanically decidable, whose only question is whether the response transmits a near-complete formal/proof implementation rather than a compact mathematical intuition.

The semantic reviewer must not judge whether the intuition is correct, elegant, likely to help Q4, or similar to a reference proof. It sees only theorem/public context + candidate intuition and returns one of:

```text
accepted_intuition
rejected_formal_implementation
rejected_near_complete_proof
rejected_not_an_intuition
```

Preserve deterministic hard reasons separately from semantic boundary decisions.

Do not sanitize/rewrite a rejected output into compliance.

# Retry policy

For each theorem:

1. generate attempt 1;
2. if it fails only the frozen output-boundary/length checks, allow one fresh regeneration with the same theorem and generator configuration plus the same frozen instruction emphasizing compact natural-language intuition;
3. maximum **2 attempts per theorem**;
4. accept the first eligible attempt;
5. if both fail, retain `missing_accepted_intuition` with raw attempts/reasons.

No retry may depend on qwen-lean, Lean, DeepSeek, Mathia, or subjective downstream usefulness.

Tool use or web use is **not** a retry trigger.

# Raw provenance and publication

For every attempt retain:

- task/workload identity;
- exact public theorem/context hash;
- exact prompt/instruction hash;
- model/runtime/reasoning configuration;
- raw output text/hash;
- output token count;
- tool-call metadata/transcript sufficient to know which tools/sources were used;
- deterministic hard-check result;
- semantic boundary-review result when invoked;
- retry/accepted-attempt identity.

The downstream accepted-intuition artifact must contain only the public task identity, accepted natural-language intuition, hashes/provenance, and evaluation-only metadata — **not tool scratchpads, browsing traces, hidden reasoning, or privileged source material**.

# Hard isolation from Mathia training

Inherit and strengthen the isolation invariant from #57.

This corpus is an external evaluation/reference artifact and must remain physically/logically separate from:

- `agnostic-mathia-full-v1`;
- `agnostic-mathia-openalex-supplement-v1`;
- Riemann-Mathia releases;
- #55 `G-v2` optimizer manifest;
- any current/future Mathia SFT/RL training manifest;
- qwen-lean/planner training targets.

Every accepted record must carry equivalent semantics to:

```text
evaluation_only: true
training_eligible: false
artifact_role: frontier_assisted_reference
```

Mechanically prove before release that:

1. no task/intuition ID is present in any trainable Mathia manifest;
2. no output path is consumed by G-v2 materialization;
3. the frozen G-v2 optimizer manifest is byte/hash-identical with this evaluation directory present or absent;
4. no #57/#new-issue frontier artifact is auto-enumerated as source/interpretation/synthesis training data.

Any contamination is a material integrity blocker.

# Required pre-full-run calibration gate

Before spending the complete 650-task frontier generation budget, publish the 24-task calibration evidence and require a fresh read-only review answering only:

- do the outputs look like compact mathematical intuitions rather than Lean plans/proofs?;
- does the boundary checker reject obvious formal implementation leakage?;
- is it avoiding the #57 false-positive pattern where ordinary mathematical/formal vocabulary caused wholesale rejection?;
- is the chosen 128/160/192 cap justified by the frozen rule?;
- are tools/web being treated as provenance rather than failures?;
- is theorem/public context still isolated from oracle proofs and qwen-lean outcomes?

Verdict:

- `CALIBRATION_PASS` -> run the remaining tasks under the frozen contract;
- `CALIBRATION_REVISE` -> revise before full generation; do not reuse calibration outputs as final corpus rows;
- `CALIBRATION_BLOCKED` -> stop.

This is deliberately a pre-compute quality gate to avoid repeating #57 at full scale with a broken extraction boundary.

# Final validation / report

After the full run report at minimum:

- exact 244 + 406 membership and hashes;
- accepted/missing counts by workload;
- first-attempt vs second-attempt acceptance;
- output-token distribution;
- deterministic rejection counts by reason;
- semantic boundary-review counts;
- tool-use rates/types and source-domain summary as provenance only;
- exact/normalized duplicate intuition rate;
- anti-contamination checks;
- fresh independent audit of the exact frozen release.

Do not rank intuitions by subjective quality and do not run qwen-lean/Lean in this issue.

# Out of scope

- source/oracle proof visibility;
- qwen-lean/Q4 inference;
- DeepSeek inference;
- Lean verification;
- Mathia-v1/v2 inference;
- planner training/inference;
- RL/GRPO;
- final-test tasks;
- filtering by Q4 difficulty;
- training Mathia on these outputs;
- claiming frontier-assisted intuition fertility before the separate downstream experiment.

# Deliverable / exit gate

Close only when one immutable `frontier-assisted-intuition-corpus-v1` release contains:

- exact 650-task validation population;
- frozen GPT-5.6 Sol/xhigh + tools-allowed generation contract;
- successful 24-task calibration gate before full generation;
- accepted natural-language mathematical intuition for every task where the max-two-attempt policy succeeds;
- raw attempt/tool provenance preserved;
- no oracle proof or qwen-lean outcome leakage;
- no final-test membership;
- hard training isolation / zero contamination;
- fresh independent audit with no material defect.

Record exactly one final decision:

- `FRONTIER_ASSISTED_INTUITION_CORPUS_READY`;
- `FRONTIER_ASSISTED_GENERATION_BLOCKER`;
- `FRONTIER_ASSISTED_OUTPUT_BOUNDARY_BLOCKER`;
- `FRONTIER_ASSISTED_INTEGRITY_BLOCKER`.

Completion authorizes only a separate preregistered downstream fertility experiment. It is not evidence by itself that the intuitions improve qwen-lean.

**Comment by murillo128** [comment]
## Amendment — full-run circuit breaker for extraction failure

Do **not** wait until all 650 tasks finish if the post-calibration extraction channel is clearly failing.

This circuit breaker applies only **after `CALIBRATION_PASS`** and does not change the frozen generator prompt, output boundary, retry policy, or task membership.

### Online acceptance checks

During the full run, compute acceptance only after each theorem has exhausted its predeclared maximum-two-attempt policy. Tool/web use is allowed and **must not** count as failure by itself.

Check the following without inspecting qwen-lean/Lean/downstream outcomes:

1. **Early 24-task check:** after the first 24 non-calibration tasks, stop if fewer than **18/24 (75%)** have an accepted intuition.
2. **Rolling 32-task checks:** thereafter, at every completed block of 32 tasks, stop if fewer than **24/32 (75%)** in that block have an accepted intuition.
3. **Consecutive-miss breaker:** stop immediately if **12 consecutive tasks** end as `missing_accepted_intuition` after both allowed attempts.
4. **Runtime-failure breaker:** in any completed 32-task block, stop if more than **8/32 (25%)** tasks end missing because of genuine generation/runtime failures. Tool calls themselves are provenance, not runtime failures.

The 75% threshold is deliberately much looser than the calibration gate (`>=22/24`) so normal task variation does not stop a healthy run; it is meant to catch a protocol/extraction collapse like #57 long before all 650 tasks are spent.

### On breaker activation

If any breaker fires:

- stop launching new frontier generations as soon as currently in-flight attempts can be safely captured;
- preserve every completed/in-flight raw attempt and all tool provenance byte-for-byte;
- publish a compact partial summary containing the triggering window/task sequence and rejection/failure reasons;
- record `FRONTIER_ASSISTED_GENERATION_BLOCKER` or `FRONTIER_ASSISTED_OUTPUT_BOUNDARY_BLOCKER` according to the observed cause;
- **do not** relax the gate, increase the token cap, alter the prompt, add retries, reorder/select tasks, or otherwise repair the running experiment after seeing the failure;
- any revised extraction contract must be frozen and restarted as a new run/version.

The circuit breaker is operational protection against wasting the remaining frontier-generation budget; it is not a quality metric and must never use mathematical correctness or downstream proof success.

**Comment by murillo128** [comment]
## Execution result — `FRONTIER_ASSISTED_INTUITION_CORPUS_READY`

Issue #59 has completed calibration, full generation, immutable finalization, publication, and fresh exact-target independent review.

- Evidence PR: #60 (ready for review; not merged)
- Exact independently reviewed corpus target: `584cc5319ecb1f1862b88bd81adbe69d1cd30000`
- Bound PASS evidence head: `9c39b5832b36af450ce94230412eb82f168ee555`
- Freeze: `frontier_assisted_intuition_corpus_26b3030e14ab4df8694b9c14ab30a297214f33f7a7ebdb8686ccd2ffae37849f`
- Population: exact 244 miniF2F + 406 fresh-composition validation tasks
- Result: 611 accepted / 39 missing, 746 attempts, frozen 128-token cap
- Breakers: none triggered; zero runtime misses; weakest complete rolling block 27/32
- Provenance: all generation and dual semantic-review captures retained; 2 web-assisted attempts recorded
- Isolation: evaluation-only, training-ineligible, zero trainable-manifest overlap or consumers
- Validation: deterministic source/manifest/final replay and all 287 repository tests pass
- Final independent audit: PASS with no material findings after all accepted outputs, retries, hard rejects, tool-assisted attempts, and prior r3 regressions were inspected

This completion authorizes only a separate preregistered downstream fertility experiment. It is not evidence of qwen-lean uplift and is not merge authorization.

Integration note: PR #60 retains its original base on the preserved #57 evidence branch; PR #58 is closed and unmerged. No base retarget, reopen, or merge was performed without explicit maintainer direction.
